content/build/hydrobuild.md

```diff
@@ -263,39 +213,112 @@ jobs:
         uses: docker/build-push-action@v4
         with:
           context: .
-          push: true
-          tags: user/app:latest
+          tags: "<org>/<image>"
+          # if this runs in a pull request, export results to build cache
+          outputs: ${{ github.event_name == 'pull_request' && 'type=cacheonly' || '' }}
+          # if this doesn't run in a pull request, push to a registry
+          push: ${{ github.event_name != 'pull_request' }}
```

Comment:
You can just use the `outputs` input:

```suggestion
          # if this runs in a pull request, export results to build cache, otherwise push to registry
          outputs: ${{ github.event_name == 'pull_request' && 'type=cacheonly' || 'type=registry,push=true' }}
```

Not sure why you need to set `type=cacheonly` though. Just the following should work:

```suggestion
          push: ${{ github.event_name != 'pull_request' }}
```

Reply:
I assume the reason for the cacheonly is because if it's got a `tag` then it does a `load` by default which isn't wanted here?

Reply:
Correct, this is to avoid `load` for tagged pull request builds. I can change this to use `outputs` instead if that's more readable.

Reply:
> I assume the reason for the cacheonly is because if it's got a `tag` then it does a `load` by default which isn't wanted here?

Hum if that's the case it's not consistent with other remote drivers (container, remote, kubernetes)

Reply:
True. The automatic pull for tags in unique for this driver, and only for tagged images.

Reply:
Indeed, the default load was added in order to facilitate inner loop flows (which was the primary focus of cloud builds when we started) which has this side effect - we need to opt out explicitly.